packages/ui/src/utils/__tests__/timePeriodUtils.test.ts

```diff
@@ -0,0 +1,331 @@
+            const result = toDHIS2PeriodData('2024-01', '2024-03', 'invalid' as keyof typeof PERIOD_TYPES);
+            expect(result).toEqual([]);
+        });
+
+        it('should return empty array for invalid month date format', () => {
+            const result = toDHIS2PeriodData('invalid', '2024-03', PERIOD_TYPES.MONTH);
+            expect(result).toEqual([]);
+        });
+
+        it('should return empty array for invalid week date format', () => {
+            const result = toDHIS2PeriodData('invalid', '2024-W03', PERIOD_TYPES.WEEK);
+            expect(result).toEqual([]);
+        });
+
+        it('should return empty array for malformed month format', () => {
+            const result = toDHIS2PeriodData('2024/01', '2024/03', PERIOD_TYPES.MONTH);
+            expect(result).toEqual([]);
+        });
+
+        it('should return empty array for malformed week format', () => {
+            const result = toDHIS2PeriodData('2024-1', '2024-3', PERIOD_TYPES.WEEK);
+            expect(result).toEqual([]);
+        });
+
+        it('should return empty array for unreasonable date range (>100 years)', () => {
+            const result = toDHIS2PeriodData('1900-01', '2100-01', PERIOD_TYPES.MONTH);
+            expect(result).toEqual([]);
+        });
+
+        it('should handle end date before start date for months', () => {
+            const result = toDHIS2PeriodData('2024-03', '2024-01', PERIOD_TYPES.MONTH);
+            expect(result).toEqual([]);
+        });
+
+        it('should handle end date before start date for weeks', () => {
+            const result = toDHIS2PeriodData('2024-W03', '2024-W01', PERIOD_TYPES.WEEK);
+            expect(result).toEqual([]);
+        });
+    });
+});
+
+describe('convertServerToClientPeriod', () => {
+    describe('month periods', () => {
+        it('should convert server month format to client format', () => {
+            expect(convertServerToClientPeriod('202401', PERIOD_TYPES.MONTH)).toBe('2024-01');
+            expect(convertServerToClientPeriod('202312', PERIOD_TYPES.MONTH)).toBe('2023-12');
+        });
+
+        it('should handle lowercase period type', () => {
+            expect(convertServerToClientPeriod('202401', PERIOD_TYPES.MONTH)).toBe('2024-01');
+        });
+
+        it('should return original value for invalid month format', () => {
+            expect(convertServerToClientPeriod('invalid', PERIOD_TYPES.MONTH)).toBe('invalid');
+        });
+    });
+
+    describe('week periods', () => {
+        it('should convert server week format to client format', () => {
+            expect(convertServerToClientPeriod('2024W01', PERIOD_TYPES.WEEK)).toBe('2024-W01');
+            expect(convertServerToClientPeriod('2024W52', PERIOD_TYPES.WEEK)).toBe('2024-W52');
+        });
+
+        it('should handle lowercase period type', () => {
+            expect(convertServerToClientPeriod('2024W01', PERIOD_TYPES.WEEK)).toBe('2024-W01');
```

Comment:
Are you sure this tests anything differently than the test on line 151? i.e., test name says "lowercase", but the second arg is the same as the previous test.